service/src/test/java/tds/assessment/repositories/impl/AdminSubjectQueryRepositoriesIntegrationTests.java

```diff
@@ -1,41 +0,0 @@
-package tds.assessment.repositories.impl;
-
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
-import org.springframework.transaction.annotation.Transactional;
-
-import java.util.Optional;
-
-import tds.assessment.SetOfAdminSubject;
-import tds.assessment.repositories.AdminSubjectQueryRepository;
-
-import static org.assertj.core.api.Assertions.assertThat;
-
-@RunWith(SpringJUnit4ClassRunner.class)
-@SpringBootTest
-@Transactional
-public class AdminSubjectQueryRepositoriesIntegrationTests {
```

Comment:
Since service methods can have conditional logic, there is a more likely case that we could miss a use-case and therefore miss testing a specific repository method that we would catch in writing repository integration test.  But I don't think this is a reason to not take this approach because I think you are right, this is double coverage really.  Just mentioning it because it makes the code coverage reports more valuable to make sure we don't miss areas.